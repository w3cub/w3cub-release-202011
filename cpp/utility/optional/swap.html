
<!DOCTYPE HTML>

<html lang="en" class="_theme-default">

<head>
  <meta charset="utf-8">
  <title>Std&#58;&#58;optional&#58;&#58;swap - C++ - W3cubDocs</title>
  
  <meta name="description" content=" Swaps the contents with those of other. ">
  <meta name="keywords" content="std, optional, t, swap, c++, cpp">
  <meta name="HandheldFriendly" content="True">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="mobile-web-app-capable" content="yes">
  
  <link rel="canonical" href="https://docs.w3cub.com/cpp/utility/optional/swap.html">
  <link href="/favicon.png" rel="icon">
  <link rel="stylesheet" type="text/css" href="/assets/application-9c7b0d969c0ce639a6c01c2badcdd81667c7c66206aa3d0fccc5cb0585fb1b912dab72ea882101d18010680927514b4b77b107282db8f756fbf63301fe9c88b0.css">
  <script type="text/javascript" src="/assets/application-79c555f6b25481fffac2cac30a7f3e54e608ca09e9e8e42bb1790095ba6d0fcace47d6bc624ddce952c70370892f2d46864f89e6943d4f7f7ff16c8a3231a91a.js"></script>
  <script src="/json/cpp.js"></script>
  
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-R3WC07G3GB"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-R3WC07G3GB');
</script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-2572770204602497"
     crossorigin="anonymous"></script>
<script async custom-element="amp-auto-ads"
  src="https://cdn.ampproject.org/v0/amp-auto-ads-0.1.js">
</script>


</head>

<body class="docs">
	<amp-auto-ads type="adsense"
              data-ad-client="ca-pub-2572770204602497">
	</amp-auto-ads>
	<div class="_app">
	<header class="_header">

  <a class="_home-link" href="/" ></a>
  <a class="_menu-link"></a>
  <h1 class="_logo">
    <a href="/" class="_nav-link" title="API Documentation Browser">W3cubDocs</a>
  </h1>
  
  <span class="_logo-sub-nav">/</span><span class="_logo-sub-nav"><a href="/cpp/" class="_nav-link" title="" style="margin-left:0;">C++</a></span>
  
  <nav class="_nav">
    <a href="https://tools.w3cub.com/?_sp=docs" target="_blank" class="_nav-link ">W3cubTools</a>
    <a href="/cheatsheets/" class="_nav-link ">Cheatsheets</a>
    <a href="/about/" class="_nav-link ">About</a>
  </nav>
</header>
	<section class="_sidebar">
		
		<form class="_search">
		  <input type="search" class="_search-input" placeholder="Search&hellip;" autocomplete="off" autocapitalize="off" autocorrect="off" spellcheck="false" maxlength="20">
		  <a class="_search-clear"></a>
		  <div class="_search-tag"></div>
		</form>
		
		<div class="_list-wrap">
			<div class="_list">
			
			</div>
		</div>
	</section>
	<section class="_container ">
		<div class="_content">
			<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-2572770204602497"
     data-ad-slot="6861657091"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
			<div class="_page _cppref">
				
				
<h1 id="firstHeading" class="firstHeading">std::optional&lt;T&gt;::swap</h1>            <table class="t-dcl-begin">   <tr class="t-dcl t-since-cxx17 t-until-cxx20"> <td> <pre data-language="cpp">void swap( optional&amp; other ) noexcept(/* see below */);</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx17">(since C++17)</span> <br><span class="t-mark-rev t-until-cxx20">(until C++20)</span> </td> </tr> <tr class="t-dcl t-since-cxx20"> <td> <pre data-language="cpp">constexpr void swap( optional&amp; other ) noexcept(/* see below */);</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx20">(since C++20)</span> </td> </tr>   </table> <p>Swaps the contents with those of <code>other</code>.</p>
<ul>
<li> If neither <code>*this</code> nor <code>other</code> contain a value, the function has no effect. </li>
<li> If only one of <code>*this</code> and <code>other</code> contains a value (let's call this object <code>in</code> and the other <code>un</code>), the contained value of <code>un</code> is <a href="../../language/direct_initialization" title="cpp/language/direct initialization">direct-initialized</a> from <code>std::move(*in)</code>, followed by destruction of the contained value of <code>in</code> as if by <code>in-&gt;T::~T()</code>. After this call, <code>in</code> does not contain a value; <code>un</code> contains a value. </li>
<li> If both <code>*this</code> and <code>other</code> contain values, the contained values are exchanged by calling <code><span class="kw1">using</span> <a href="http://en.cppreference.com/w/cpp/algorithm/swap"><span class="kw1604">std::<span class="me2">swap</span></span></a><span class="sy4">;</span> swap<span class="br0">(</span><span class="sy2">**</span>this, <span class="sy2">*</span>other<span class="br0">)</span></code>. <code>T</code> lvalues must satisfy <a href="../../named_req/swappable" title="cpp/named req/Swappable">Swappable</a>. </li>
</ul>   <p>The program is ill-formed if <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">std::<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> is <code>false</code>.</p>
<h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> other </td> <td> - </td> <td> the <code>optional</code> object to exchange the contents with </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p>(none).</p>
<h3 id="Exceptions"> Exceptions</h3> <div class="t-noexcept-full">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a> specification: <div class="t-noexcept-box"><span class="t-cc"><code><span class="kw1">noexcept</span><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span> <span class="sy3">&amp;&amp;</span> <br> <a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw580">std::<span class="me2">is_nothrow_swappable_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span><span class="br0">)</span></code></span></div>
</div> <p>In the case of thrown exception, the states of the contained values of <code>*this</code> and <code>other</code> are determined by the exception safety guarantees of <code>swap</code> of type <code>T</code> or <code>T</code>'s move constructor, whichever is called. For both <code>*this</code> and <code>other</code>, if the object contained a value, it is left containing a value, and the other way round.</p>
<h3 id="Example"> Example</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">#include &lt;iostream&gt;
#include &lt;string&gt;
#include &lt;optional&gt;
 
int main()
{
    std::optional&lt;std::string&gt; opt1("First example text");
    std::optional&lt;std::string&gt; opt2("2nd text");
 
    enum Swap { Before, After };
    auto print_opts = [&amp;](Swap e) {
        std::cout &lt;&lt; (e == Before ? "Before swap:\n" : "After swap:\n");
        std::cout &lt;&lt; "opt1 contains '" &lt;&lt; opt1.value_or("") &lt;&lt; "'\n";
        std::cout &lt;&lt; "opt2 contains '" &lt;&lt; opt2.value_or("") &lt;&lt; "'\n";
        std::cout &lt;&lt; (e == Before ? "---SWAP---\n": "\n");
    };
 
    print_opts(Before);
    opt1.swap(opt2);
    print_opts(After);
 
    // Swap with only 1 set
    opt1 = "Lorem ipsum dolor sit amet, consectetur tincidunt.";
    opt2.reset();
 
    print_opts(Before);
    opt1.swap(opt2);
    print_opts(After);
}</pre></div> <p>Output:</p>
<div class="text source-text"><pre data-language="c">Before swap:
opt1 contains 'First example text'
opt2 contains '2nd text'
---SWAP---
After swap:
opt1 contains '2nd text'
opt2 contains 'First example text'
 
Before swap:
opt1 contains 'Lorem ipsum dolor sit amet, consectetur tincidunt.'
opt2 contains ''
---SWAP---
After swap:
opt1 contains ''
opt2 contains 'Lorem ipsum dolor sit amet, consectetur tincidunt.'</pre></div> </div> <h3 id="Defect_reports"> Defect reports</h3> <p>The following behavior-changing defect reports were applied retroactively to previously published C++ standards.</p>
<table class="dsctable"> <tr> <th> DR </th> <th> Applied to </th> <th> Behavior as published </th> <th> Correct behavior </th>
</tr> <tr> <td> <a rel="nofollow" class="external text" href="https://wg21.link/P2231R1">P2231R1</a> </td> <td> C++20 </td> <td> <code>swap</code> was not constexpr while the required operations can be constexpr in C++20 </td> <td> made constexpr </td>
</tr>
</table> <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="swap2" title="cpp/utility/optional/swap2"> <span class="t-lines"><span>std::swap<span class="t-dsc-small">(std::optional)</span></span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx17">(C++17)</span></span></span></div> </td> <td> specializes the <code><a href="../../algorithm/swap" title="cpp/algorithm/swap">std::swap</a></code> algorithm <br> <span class="t-mark">(function template)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    Â© cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/utility/optional/swap" class="_attribution-link">https://en.cppreference.com/w/cpp/utility/optional/swap</a>
  </p>
</div>

				
			</div>
			<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-2572770204602497"
     data-ad-slot="1992473792"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
		</div>
	</section>

	</div>
	<svg style="display:none">
		<symbol id="icon-dir"><svg viewBox="0 0 20 20"><path d="M15 10c0 .3-.305.515-.305.515l-8.56 5.303c-.625.41-1.135.106-1.135-.67V4.853c0-.777.51-1.078 1.135-.67l8.56 5.305S15 9.702 15 10z"/></svg></symbol>
	  </svg>
</body>
</html>
