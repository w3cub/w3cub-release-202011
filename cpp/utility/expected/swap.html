
<!DOCTYPE HTML>

<html lang="en" class="_theme-default">

<head>
  <meta charset="utf-8">
  <title>Std&#58;&#58;expected&#58;&#58;swap - C++ - W3cubDocs</title>
  
  <meta name="description" content=" Swaps the contents with those of other. ">
  <meta name="keywords" content="std, expected, t, e, swap, c++, cpp">
  <meta name="HandheldFriendly" content="True">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="mobile-web-app-capable" content="yes">
  
  <link rel="canonical" href="https://docs.w3cub.com/cpp/utility/expected/swap.html">
  <link href="/favicon.png" rel="icon">
  <link rel="stylesheet" type="text/css" href="/assets/application-9c7b0d969c0ce639a6c01c2badcdd81667c7c66206aa3d0fccc5cb0585fb1b912dab72ea882101d18010680927514b4b77b107282db8f756fbf63301fe9c88b0.css">
  <script type="text/javascript" src="/assets/application-79c555f6b25481fffac2cac30a7f3e54e608ca09e9e8e42bb1790095ba6d0fcace47d6bc624ddce952c70370892f2d46864f89e6943d4f7f7ff16c8a3231a91a.js"></script>
  <script src="/json/cpp.js"></script>
  
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-R3WC07G3GB"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-R3WC07G3GB');
</script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-2572770204602497"
     crossorigin="anonymous"></script>
<script async custom-element="amp-auto-ads"
  src="https://cdn.ampproject.org/v0/amp-auto-ads-0.1.js">
</script>


</head>

<body class="docs">
	<amp-auto-ads type="adsense"
              data-ad-client="ca-pub-2572770204602497">
	</amp-auto-ads>
	<div class="_app">
	<header class="_header">

  <a class="_home-link" href="/" ></a>
  <a class="_menu-link"></a>
  <h1 class="_logo">
    <a href="/" class="_nav-link" title="API Documentation Browser">W3cubDocs</a>
  </h1>
  
  <span class="_logo-sub-nav">/</span><span class="_logo-sub-nav"><a href="/cpp/" class="_nav-link" title="" style="margin-left:0;">C++</a></span>
  
  <nav class="_nav">
    <a href="https://tools.w3cub.com/?_sp=docs" target="_blank" class="_nav-link ">W3cubTools</a>
    <a href="/cheatsheets/" class="_nav-link ">Cheatsheets</a>
    <a href="/about/" class="_nav-link ">About</a>
  </nav>
</header>
	<section class="_sidebar">
		
		<form class="_search">
		  <input type="search" class="_search-input" placeholder="Search&hellip;" autocomplete="off" autocapitalize="off" autocorrect="off" spellcheck="false" maxlength="20">
		  <a class="_search-clear"></a>
		  <div class="_search-tag"></div>
		</form>
		
		<div class="_list-wrap">
			<div class="_list">
			
			</div>
		</div>
	</section>
	<section class="_container ">
		<div class="_content">
			<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-2572770204602497"
     data-ad-slot="6861657091"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
			<div class="_page _cppref">
				
				
<h1 id="firstHeading" class="firstHeading">std::expected&lt;T,E&gt;::swap</h1>            <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr void swap( expected&amp; other ) noexcept(/*see below*/);</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr>  </table> <p>Swaps the contents with those of <code>other</code>.</p>
<ul>
<li> If both <a href="operator_bool" title="cpp/utility/expected/operator bool"><code>this-&gt;has_value()</code></a> and <code>other.has_value()</code> are true: </li>
<ul>
<li> If <code>T</code> is (possibly cv-qualified) <code>void</code>, no effects. </li>
<li> Otherwise, equivalent to <code><span class="kw1">using</span> <a href="http://en.cppreference.com/w/cpp/algorithm/swap"><span class="kw1604">std::<span class="me2">swap</span></span></a><span class="sy4">;</span> swap<span class="br0">(</span><span class="sy2">*</span>this, other<span class="br0">)</span><span class="sy4">;</span></code>. </li>
</ul>
<li> If both <code>this-&gt;has_value()</code> and <code>other.has_value()</code> are false, equivalent to<br><code><span class="kw1">using</span> <a href="http://en.cppreference.com/w/cpp/algorithm/swap"><span class="kw1604">std::<span class="me2">swap</span></span></a><span class="sy4">;</span> swap<span class="br0">(</span>this<span class="sy2">-</span><span class="sy1">&gt;</span>error<span class="br0">(</span><span class="br0">)</span>, other.<span class="me1">error</span><span class="br0">(</span><span class="br0">)</span><span class="br0">)</span><span class="sy4">;</span></code>. </li>
<li> If <code>this-&gt;has_value()</code> is false and <code>other.has_value()</code> is true, calls <code>other.swap(*this)</code>. </li>
<li> If <code>this-&gt;has_value()</code> is true and <code>other.has_value()</code> is false, </li>
<ul><li> If <code>T</code> is (possibly cv-qualified) <code>void</code>, let <i>unex</i> be the member that represents the unexpected value, equivalent to: </li></ul>
</ul>    <div class="cpp source-cpp"><pre data-language="cpp">std::construct_at(std::addressof(unex), std::move(other.unex));
std::destroy_at(std::addressof(other.unex));</pre></div> <ul><li> Otherwise, let <i>val</i> be the member that represents the expected value and <i>unex</i> be the member that represents the unexpected value, equivalent to: </li></ul> <div class="cpp source-cpp"><pre data-language="cpp">if constexpr (std::is_nothrow_move_constructible_v&lt;E&gt;) {
    E temp(std::move(other.unex));
    std::destroy_at(std::addressof(other.unex));
    try {
        std::construct_at(std::addressof(other.val), std::move(val));
        std::destroy_at(std::addressof(val));
        std::construct_at(std::addressof(unex), std::move(temp));
    } catch(...) {
        std::construct_at(std::addressof(other.unex), std::move(temp));
        throw;
    }
} else {
    T temp(std::move(val));
    std::destroy_at(std::addressof(val));
    try {
        std::construct_at(std::addressof(unex), std::move(other.unex));
        std::destroy_at(std::addressof(other.unex));
        std::construct_at(std::addressof(other.val), std::move(temp));
    } catch(...) {
        std::construct_at(std::addressof(val), std::move(temp));
        throw;
    }
}</pre></div> <ul><li> In either case, if no exception was thrown, after swap, <code>this-&gt;has_value()</code> is false, and <code>other.has_value()</code> is true. </li></ul> <p>This function participates in overload resolution only if.</p>
<ul>
<li> either <code>T</code> is (possibly cv-qualified) <code>void</code>, or <code><a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw576">std::<span class="me2">is_swappable_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> is true, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw576">std::<span class="me2">is_swappable_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code> is true, and </li>
<li> either <code>T</code> is (possibly cv-qualified) <code>void</code>, or <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">std::<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> is true, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">std::<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code> is true, and </li>
<li> at least one of the following is true: <ul>
<li> <code>T</code> is (possibly cv-qualified) <code>void</code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code> </li>
</ul> </li>
</ul>  <h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> other </td> <td> - </td> <td> the <code>optional</code> object to exchange the contents with </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p>(none).</p>
<h3 id="Exceptions"> Exceptions</h3> If <code>T</code> is (possibly cv-qualified) <code>void</code>, <div class="t-noexcept-full">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a> specification: <div class="t-noexcept-box"><span class="t-cc"><code><span class="kw1">noexcept</span><span class="br0">(</span><br> <p><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span> <span class="sy3">&amp;&amp;</span> <a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw580">std::<span class="me2">is_nothrow_swappable_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span><br></p>
<span class="br0">)</span></code></span></div>
</div> Otherwise, <div class="t-noexcept-full">
<a href="../../language/noexcept_spec" title="cpp/language/noexcept spec"><code>noexcept</code></a> specification: <div class="t-noexcept-box"><span class="t-cc"><code><span class="kw1">noexcept</span><span class="br0">(</span><br> <p><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span> <span class="sy3">&amp;&amp;</span> <a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw580">std::<span class="me2">is_nothrow_swappable_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span> <span class="sy3">&amp;&amp;</span><br> <a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span> <span class="sy3">&amp;&amp;</span> <a href="http://en.cppreference.com/w/cpp/types/is_swappable"><span class="kw580">std::<span class="me2">is_nothrow_swappable_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span><br></p>
<span class="br0">)</span></code></span></div>
</div> <p>In the case of thrown exception, the states of the contained values of <code>*this</code> and <code>other</code> are determined by the exception safety guarantees of <code>swap</code> or <code>T</code>'s and <code>E</code>'s move constructor, whichever is called. For both <code>*this</code> and <code>other</code>, if the object contained an expected value, it is left containing an expected value, and the other way round.</p>
<h3 id="Example"> Example</h3>  <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="swap2" title="cpp/utility/expected/swap2"> <span class="t-lines"><span>swap<span class="t-dsc-small">(std::expected)</span></span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx23">(C++23)</span></span></span></div> </td> <td> specializes the <code><a href="../../algorithm/swap" title="cpp/algorithm/swap">std::swap</a></code> algorithm <br> <span class="t-mark">(function)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    © cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/utility/expected/swap" class="_attribution-link">https://en.cppreference.com/w/cpp/utility/expected/swap</a>
  </p>
</div>

				
			</div>
			<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-2572770204602497"
     data-ad-slot="1992473792"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
		</div>
	</section>

	</div>
	<svg style="display:none">
		<symbol id="icon-dir"><svg viewBox="0 0 20 20"><path d="M15 10c0 .3-.305.515-.305.515l-8.56 5.303c-.625.41-1.135.106-1.135-.67V4.853c0-.777.51-1.078 1.135-.67l8.56 5.305S15 9.702 15 10z"/></svg></symbol>
	  </svg>
</body>
</html>
